src/dataLoader/savedItemsDataLoader.ts

```diff
@@ -0,0 +1,51 @@
+import DataLoader from 'dataloader';
+import { SavedItemDataService } from '../dataService/queryServices';
+import { IContext } from '../server/context';
+import { SavedItem } from '../types';
+
+/**
+ * Reorders saved items based on the URLs that are received to
+ * keep the response in the same order of urls as requested
+ * by the Apollo Gateway (Client API)
```

Comment:
Ah I see, this is because the SQL query response doesn't care about ordering.

Reply:
Collections had a similar bug. https://pocket.slack.com/archives/CTLKW5WCF/p1638548978165400

We have to be careful when using a dataloader with reference resolvers